apps/sms/services/js/moz_mobile_message/messaging_service.js

```diff
@@ -0,0 +1,193 @@
+}
+
+var MessagingService = {
+  /**
+   * Initializes our service using the Service mixin's initService and
+   * requests mozMobileMessage shim as client.
+   */
+  init() {
+    ensureBridge();
+
+    this.initService();
+
+    this.mozMobileMessageClient = bridge.client(MOBILE_MESSAGE_CLIENT_NAME);
+  },
+
+  /**
+   * Send sms by using send API shim.
+   * @param {*} opts Options for sending sms contains following properties:
+   * - recipients (string or array of string): contains the list of
+   *   recipients for this message.
+   * - content (string): the message's body.
+   * - serviceId (optional long or string): the SIM serviceId we use to send
+   *   the message.
+   * @returns {Promise} Promise with array of the request result.
+   */
+  sendSMS(opts) {
+    var recipients = opts.recipients || [],
+        content = opts.content,
+        serviceId = sanitizeServiceId(opts.serviceId);
+
+    if (!Array.isArray(recipients)) {
+      recipients = [recipients];
+    }
+
+    var sendOpts = getSendOptionsFromServiceId(serviceId);
+    var results = recipients.map(
+      (recipient) => this.mozMobileMessageClient(
+        'send', recipient, content, sendOpts
+      ).then((evt) => evt, (evt) => evt)
+    );
+
+    return Promise.all(results).then((values) =>
+      values.map((event, idx) => ({
+        success: !event.target.error,
+        result: event.target.result,
+        code: event.target.error,
+        recipient: recipients[idx]
+      }))
+    );
+
+  },
+
+  /**
+   * Send mms by using sendMMS API shim.
+   * @param {*} opts Options for sending mms contains following properties:
+   * - recipients (string or array of string): recipients for this message.
+   * - subject (optional string): subject for this message.
+   * - content (array of SMIL slides): this is the content for the message (
+   *   see ConversationView for more information).
+   * - serviceId (optional long or string): the SIM that should be used for
+   *   sending this message. If this is not the current default configuration
+   *   for sending MMS, then we'll first switch the configuration to this
+   *   serviceId, and only then send the message. That means that the
+   *   "sending" event will come quite late in this case.
+   * @returns {Promise} DOMRequest promise with success result or error.
+   */
+  sendMMS(opts) {
+    var recipients = opts.recipients,
+        subject = opts.subject,
+        content = opts.content,
+        serviceId = opts.serviceId = sanitizeServiceId(opts.serviceId);
+
+    if (!Array.isArray(recipients)) {
+      recipients = [recipients];
+    }
+
+    var message = SMIL.generate(content);
+
+    var sendOpts = getSendOptionsFromServiceId(serviceId);
+
+    return this.mozMobileMessageClient('sendMMS', {
+      receivers: recipients,
+      subject: subject,
+      smil: message.smil,
+      attachments: message.attachments
+    }, sendOpts).then(
+      (evt) => evt.target.result,
+      (evt) => {
```

Comment:
I think ideally - yes, but I'd rather do it later as enhancement/continuous-improvement once this service works and used by at least by one "real" consumer so that we can do smart things easier :)